core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -513,41 +495,41 @@ public V replace(K key, V value) {
 
    @Override
    public void addListener(Object listener) {
-      cacheImplementation.notifier.addListener(listener, classLoader.get());
+      cacheImplementation.notifier.addListener(listener, (ClassLoader) null);
    }
 
    @Override
    public void addListener(Object listener, KeyFilter<? super K> filter) {
-      cacheImplementation.notifier.addListener(listener, filter, classLoader.get());
+      cacheImplementation.notifier.addListener(listener, filter, null);
```

Comment:
I was more asking if we wanted to remove/deprecate these, not actually remove them right now, since this is a bit different then storing entries. To be honest I never quite understood the use case for the listener myself.  But it sounds like there is no discussion required then and we can just deprecate/remove.